### ISSUE TITLE
Implement Detailed file statistics capture

### ISSUE THREAD

**Issue by PeterZaitsev** [prometheus/mysqld_exporter#126]
Hi,

Currently there is "event name" based file statistic implemented which allows to see the kind of IO but not towards what file it goes.  For example innodb_data_file will be used both for main tablespace and individual tables which have IO. 

file_summary_by_event_name

provides more detailed information showing actual file name.

**************************\* 4. row ***************************
                FILE_NAME: /var/lib/mysql/ibdata1
               EVENT_NAME: wait/io/file/innodb/innodb_data_file
    OBJECT_INSTANCE_BEGIN: 140501553632064
               COUNT_STAR: 2522
           SUM_TIMER_WAIT: 59152857149693
           MIN_TIMER_WAIT: 434922
           AVG_TIMER_WAIT: 23454740938
           MAX_TIMER_WAIT: 4821765466863
               COUNT_READ: 246
           SUM_TIMER_READ: 1295329342336
           MIN_TIMER_READ: 26980248
           AVG_TIMER_READ: 5265566296
           MAX_TIMER_READ: 397393276998
 SUM_NUMBER_OF_BYTES_READ: 6144000
              COUNT_WRITE: 1718
          SUM_TIMER_WRITE: 267658895693
          MIN_TIMER_WRITE: 2211063
          AVG_TIMER_WRITE: 155796770
          MAX_TIMER_WRITE: 115813693743
SUM_NUMBER_OF_BYTES_WRITE: 28147712
               COUNT_MISC: 558
           SUM_TIMER_MISC: 57589868911664
           MIN_TIMER_MISC: 434922
           AVG_TIMER_MISC: 103207650106
           MAX_TIMER_MISC: 4821765466863
**************************\* 5. row ***************************

**Comment by SuperQ** [comment]
Oh, so you just want to have the `FILE_NAME` column as a label for this collector

https://github.com/prometheus/mysqld_exporter/blob/master/collector/perf_schema_file_events.go#L11-L17

**Comment by PeterZaitsev** [comment]
Ben,

Somewhat.  I do not think that particular table has information about file
name:

mysql> select * from performance_schema.file_summary_by_event_name limit 1
\G
*************************** 1. row ***************************
               EVENT_NAME: wait/io/file/sql/map
               COUNT_STAR: 0
           SUM_TIMER_WAIT: 0
           MIN_TIMER_WAIT: 0
           AVG_TIMER_WAIT: 0
           MAX_TIMER_WAIT: 0
               COUNT_READ: 0
           SUM_TIMER_READ: 0
           MIN_TIMER_READ: 0
           AVG_TIMER_READ: 0
           MAX_TIMER_READ: 0
 SUM_NUMBER_OF_BYTES_READ: 0
              COUNT_WRITE: 0
          SUM_TIMER_WRITE: 0
          MIN_TIMER_WRITE: 0
          AVG_TIMER_WRITE: 0
          MAX_TIMER_WRITE: 0
SUM_NUMBER_OF_BYTES_WRITE: 0
               COUNT_MISC: 0
           SUM_TIMER_MISC: 0
           MIN_TIMER_MISC: 0
           AVG_TIMER_MISC: 0
           MAX_TIMER_MISC: 0
1 row in set (0.00 sec)


Hence different performance schema table to query from



On Sun, May 8, 2016 at 12:04 PM, Ben Kochie <notifications@github.com>
wrote:

> Oh, so you just want to have the FILE_NAME column as a label for this
> collector
>
>
> https://github.com/prometheus/mysqld_exporter/blob/master/collector/perf_schema_file_events.go#L11-L17
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/prometheus/mysqld_exporter/issues/126#issuecomment-217729725>
>



-- 
Peter Zaitsev, CEO, Percona
Tel: +1 888 401 3401 ext 7360   Skype:  peter_zaitsev

**Comment by SuperQ** [comment]
Yea, I was just reading the 5.6/5.7 documentation, the `file_summary_by_event_name` table does not include the filename.  Are you talking about using `file_summary_by_instance` instead?

**Comment by PeterZaitsev** [comment]
Yes.

I mentioned it in the report, but might be was not to clear.

I would actually keep them both as an option.  If someone has 10.000+
tables with partitions it might be too expensive to capture/keep IO stats
for every file but it will work on event name basics.

On Sun, May 8, 2016 at 12:10 PM, Ben Kochie <notifications@github.com>
wrote:

> Yea, I was just reading the 5.6/5.7 documentation, the
> file_summary_by_event_name table does not include the filename. Are you
> talking about using file_summary_by_instance instead?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/prometheus/mysqld_exporter/issues/126#issuecomment-217730106>
>



-- 
Peter Zaitsev, CEO, Percona
Tel: +1 888 401 3401 ext 7360   Skype:  peter_zaitsev

**Comment by SuperQ** [comment]
Yea, agreed, 10k+ gets into label cardinality limits.  This should be simple enough to implement as a separate collector.

```
> select count(*) from performance_schema.file_summary_by_instance;
431
> select count(*) from performance_schema.file_summary_by_event_name;
47
```

**Comment by SuperQ** [comment]
I think this is now fixed with #189.
